src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -302,60 +331,55 @@ std::shared_ptr<Channel> TwitchIrcServer::getChannelOrEmptyByID(
     return Channel::getEmpty();
 }
 
-namespace {
-    // TODO: combine this with getEmoteSetBatches in TwitchAccount.cpp, maybe some templated thing
-    std::vector<QStringList> getChannelsInBatches(QStringList channels)
-    {
-        constexpr int batchSize = 100;
-
-        int batchCount = (channels.size() / batchSize) + 1;
-
-        std::vector<QStringList> batches;
-        batches.reserve(batchCount);
+void TwitchIrcServer::bulkRefreshLiveStatus()
+{
+    auto twitchChans = std::make_shared<QHash<QString, TwitchChannel *>>();
 
-        for (int i = 0; i < batchCount; i++)
+    this->forEachChannel([twitchChans](ChannelPtr chan) {
+        auto tc = dynamic_cast<TwitchChannel *>(chan.get());
+        if (tc && !tc->roomId().isEmpty())
         {
-            QStringList batch;
-
-            // I hate you, msvc
-            int last = (std::min)(batchSize, channels.size() - batchSize * i);
-            for (int j = 0; j < last; j++)
-            {
-                batch.push_back(channels.at(j + (batchSize * i)));
-            }
-            batches.emplace_back(batch);
+            twitchChans->insert(tc->roomId(), tc);
         }
-
-        return batches;
-    }
-}  // namespace
-
-void TwitchIrcServer::bulkRefreshLiveStatus()
-{
-    QStringList userIDs;
-    this->forEachChannel([&userIDs](ChannelPtr chan) {
-        auto twitchChan = dynamic_cast<TwitchChannel *>(chan.get());
-        if (!twitchChan->roomId().isEmpty())
-            userIDs.push_back(twitchChan->roomId());
     });
 
-    for (const auto &batch : getChannelsInBatches(userIDs))
+    // iterate over batches of channel IDs
+    for (const auto &batch : getChannelsInBatches(twitchChans->keys()))
     {
         getHelix()->fetchStreams(
-            batch, QStringList(),
-            [this](std::vector<HelixStream> streams) {
+            batch, {},
+            [twitchChans](std::vector<HelixStream> streams) {
                 for (const auto &stream : streams)
                 {
-                    auto chan = this->getChannelOrEmpty(stream.userLogin);
-                    if (chan->getType() != Channel::Type::Twitch)
+                    // remaining channels will be used later to set their stream status as offline
+                    // so we use take(id) to remove it
+                    auto tc = twitchChans->take(stream.userId);
```

Comment:
Aside: if a channel changes their username after roomId has been set, and goes live, we could detect the new `userLogin` here & possibly join the new channel name

Reply:
Good to keep in mind for the future :+1: although at the moment we have 0 logic handling renames so while this is potentially useful, it'll have to be deferred until unforeseen future.